packages/design-system/src/scalars/components/number-field/number-field.tsx

```diff
@@ -66,7 +68,23 @@ export const NumberFieldRaw = forwardRef<HTMLInputElement, NumberFieldProps>(
       trailingZeros,
       precision,
     );
+    const handleChange = (
+      e: React.MouseEvent<HTMLButtonElement>,
+      operation: "increment" | "decrement",
+    ) => {
+      e.preventDefault();
+      const currentValue = Number(displayValue || defaultValue || 0);
+      const adjustment = (operation === "increment" ? 1 : -1) * (step || 1);
+      let newValue = currentValue + adjustment;
 
+      if (maxValue !== undefined && newValue > maxValue) {
+        newValue = maxValue;
+      } else if (minValue !== undefined && newValue < minValue) {
+        newValue = minValue;
+      }
+
+      onChange(newValue as unknown as React.ChangeEvent<HTMLInputElement>);
```

Comment:
You should create an event, no casting it, even when TS accept it because you're forcing it type, it may have error in other places where the onChange is expecting a real event

Reply:
you are right, done